test/AdoNetCore.AseClient.Tests/Unit/MessageTokenHandlerTests.cs

```diff
@@ -42,5 +42,52 @@ public void AssertNoErrors_EmitsAseException_WithErrorsInOrderOfSeverity()
             var ex = Assert.Throws<AseException>(() => handler.AssertNoErrors());
             Assert.AreEqual("Most Severe", ex.Message);
         }
+
+        [Test]
+        public void AssertCanDetermineBackupErrors()
+        {
+            var handler = new MessageTokenHandler();
+            handler.Handle(new EedToken
+            {
+                Severity = 1,
+                ServerName = "AZ3_BS",
+                LineNumber = 0,
+                Message = "Backup Server: 4.172.1.4: The value of 'allocated pages threshold' has been set to 40%.\n", // This is severity 1, i.e. informational
+                MessageNumber = 417201,
+                ProcedureName = "bs_optimize",
+                SqlState = new byte[0],
+                State = 40,
+                Status = EedToken.EedStatus.TDS_NO_EED,
+                TransactionStatus = TranState.TDS_NOT_IN_TRAN
+            });
+            handler.Handle(new EedToken
+            {
+                Severity = 2, // This is not >10, but severity "2" in the backup server's message, so must be treated as error. 
+                ServerName = "AZ3_BS",
+                LineNumber = 0,
+                Message = "Backup Server: 4.141.2.40: [11] The 'open' call failed for database/archive device while working on stripe device '/doesnotexist/foo' with error number 2 (No such file or directory). Refer to your operating system documentation for further details.\0",
+                MessageNumber = 414102,
+                ProcedureName = "bs_write_header",
+                SqlState = new byte[0],
+                State = 40,
+                Status = EedToken.EedStatus.TDS_NO_EED,
+                TransactionStatus = TranState.TDS_NOT_IN_TRAN
+            });
+            handler.Handle(new EedToken
+            {
+                Severity = 16,
+                ServerName = "AZ3",
+                LineNumber = 1,
+                Message = "Error encountered by Backup Server.  Please refer to Backup Server messages for details.\n",
+                MessageNumber = 8009,
+                ProcedureName = "",
+                SqlState = new byte[0],
+                State = 1,
+                Status = EedToken.EedStatus.TDS_EED_INFO,
+                TransactionStatus = TranState.TDS_NOT_IN_TRAN
+            });
+            var ex = Assert.Throws<AseException>(() => handler.AssertNoErrors());
+            Assert.AreEqual(2, ex.Errors.Count);
```

Comment:
Shouldn't `ex.Errors.Count` be `3`?

Also, I'd consider writing an Integration test which runs the test against this driver, as well as the reference driver. It'll help in determining if there are differences in the outcomes.

Check out how this is done in [ImageTests.cs](https://github.com/DataAction/AdoNetCore.AseClient/blob/6283e57aa58194e74e2ddf1713e27b2906566e08/test/AdoNetCore.AseClient.Tests/Integration/Query/ImageTests.cs#L11)
If the test gets run in .net 4.6, then the test will be run against the reference driver as well as this driver. That way, discrepancies can be detected.